File: Plus One/Plus One.java
public class Solution {
    public int[] plusOne(int[] digits) {
        int shift = 1;
        int[] res;
        for (int i = digits.length - 1; i >= 0; --i) {
            int sum = digits[i] + shift;
            shift = sum / 10;
            digits[i] = sum % 10;
        }
        if (shift != 0) {
            res = new int[digits.length + 1];
            System.arraycopy(digits, 0, res, 1, digits.length);
            res[0] = shift;
        } else {
            res = new int[digits.length];
            System.arraycopy(digits, 0, res, 0, digits.length);
        }
        return res;
    }
}

File: Plus One/Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int> &digits) {
        int shift = 1;
        for (int i = digits.size() - 1; i >= 0; --i) {
            if (shift == 0) break;
            int res = digits[i] + shift;
            shift = res / 10;
            digits[i] = res % 10;
        }
        if (shift != 0) {
            digits.insert(digits.begin(), shift);
        }
        return digits;
    }
};